MobiusCore/Source/EffectHandlers/EffectRouter.swift

```diff
@@ -98,24 +101,40 @@ public struct _PartialEffectRouter<Effect, EffectParameters, Event> {
         _ connectable: C
     ) -> EffectRouter<Effect, Event> where C.Input == EffectParameters, C.Output == Event {
         let connectable = ThreadSafeConnectable(connectable: connectable)
-        let route = Route(extractParameters: path, connectable: connectable)
+        let route = Route(extractParameters: path, connectable: connectable, queue: queue)
         return EffectRouter(routes: routes + [route])
     }
+
+    /// Handle an the current `Effect` asynchronously on the provided `DispatchQueue`
+    ///
+    /// - Parameter queue: The `DispatchQueue` that the current `Effect` should be handled on.
+    public func on(_ queue: DispatchQueue) -> Self {
```

Comment:
This is currently returning `Self`, which means that `.on(..)` can be called multiple times on the same route.

This could be solved by adding another intermediate type. I'm not sure this is worth it.

We could also crash if a `queue` has already been set. Again, I think the risk of this is quite low, but we can discuss this further here in the comments.